src/pages/RFOX/components/RewardsAndClaims/Rewards.tsx

```diff
@@ -1,32 +1,126 @@
 import { Box, CardBody } from '@chakra-ui/react'
-import { thorchainAssetId } from '@shapeshiftoss/caip'
-import { TxStatus } from '@shapeshiftoss/unchained-client'
+import type { AccountId } from '@shapeshiftoss/caip'
+import { fromAccountId, thorchainAssetId, thorchainChainId, toAccountId } from '@shapeshiftoss/caip'
+import { DAO_TREASURY_THORCHAIN } from '@shapeshiftoss/utils'
+import { getConfig } from 'config'
+import { uniq } from 'lodash'
 import { useMemo } from 'react'
+import { Text } from 'components/Text'
 import { TransactionsGroupByDate } from 'components/TransactionHistory/TransactionsGroupByDate'
-import { selectTxIdsByFilter } from 'state/slices/selectors'
+import { isSome } from 'lib/utils'
+import { selectRuneAddress } from 'pages/RFOX/helpers'
+import { useStakingInfoHistoryQuery } from 'pages/RFOX/hooks/useStakingInfoHistoryQuery'
+import { useStakingInfoQuery } from 'pages/RFOX/hooks/useStakingInfoQuery'
+import { selectReceivedTxsForAccountIdsByFilter } from 'state/slices/selectors'
 import { useAppSelector } from 'state/store'
 
+type RewardsContentProps = {
+  stakingAssetAccountId: AccountId
+}
+
 type RewardsProps = {
+  stakingAssetAccountId: AccountId
   headerComponent: JSX.Element
 }
 
-export const Rewards = ({ headerComponent }: RewardsProps) => {
-  const filter = useMemo(
-    () => ({ txStatus: TxStatus.Confirmed, assetIdFilter: thorchainAssetId }),
-    [],
+const RewardsContent = ({ stakingAssetAccountId }: RewardsContentProps) => {
+  const stakingAssetAccountAddress = useMemo(
+    () => fromAccountId(stakingAssetAccountId).account,
+    [stakingAssetAccountId],
   )
-  const txIds = useAppSelector(state => selectTxIdsByFilter(state, filter))
+
+  const {
+    data: historicalRuneAddresses,
+    isLoading: isStakingInfoHistoryLoading,
+    isFetching: isStakingInfoHistoryFetching,
+  } = useStakingInfoHistoryQuery({ stakingAssetAccountAddress, select: selectRuneAddress })
+
+  const {
+    data: currentRuneAddress,
+    isLoading: isCurrentStakingInfoLoading,
+    isFetching: isCurrentStakingInfoFetching,
+  } = useStakingInfoQuery({ stakingAssetAccountAddress, select: selectRuneAddress })
+
+  const uniqueHistoricalRuneAddresses = useMemo(() => {
+    if (!historicalRuneAddresses) {
+      return []
+    }
+
+    return uniq(historicalRuneAddresses.filter(isSome))
+  }, [historicalRuneAddresses])
+
+  const runeAddresses = useMemo(() => {
+    if (!currentRuneAddress) {
+      return uniqueHistoricalRuneAddresses
+    }
+
+    return uniq([...uniqueHistoricalRuneAddresses, currentRuneAddress])
```

Comment:
question: same question about the selector

Reply:
This one is combining output from 2 different hooks so `select` isn't applicable here